hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/snapshot/OMSnapshotCreateResponse.java

```diff
@@ -68,5 +74,25 @@ public void addToDBBatch(OMMetadataManager omMetadataManager,
     // Add to db
     omMetadataManager.getSnapshotInfoTable().putWithBatch(batchOperation,
         key, snapshotInfo);
+
+    // TODO: [SNAPSHOT] Move to createOmSnapshotCheckpoint and add table lock
+    // Remove all entries from renamedKeyTable
+    TableIterator<String, ? extends Table.KeyValue<String, OmKeyRenameInfo>>
+        iterator = omMetadataManager.getRenamedKeyTable().iterator();
+
+    String dbSnapshotBucketKey = omMetadataManager.getBucketKey(
+        snapshotInfo.getVolumeName(), snapshotInfo.getBucketName())
+        + OM_KEY_PREFIX;
+    iterator.seek(dbSnapshotBucketKey);
+
+    while (iterator.hasNext()) {
+      String renameDbKey = iterator.next().getKey();
+      if (!renameDbKey.contains(dbSnapshotBucketKey)) {
```

Comment:
Should use `startsWith` ?

```suggestion
      if (!renameDbKey.startsWith(dbSnapshotBucketKey)) {
```